package/MDAnalysis/lib/util.py

```diff
@@ -214,6 +214,8 @@
 from .picklable_file_io import pickle_open, bz2_pickle_open, gzip_pickle_open
 
 from ..exceptions import StreamWarning, DuplicateWarning
+import MDAnalysis as mda
```

Comment:
Since you only use the `AtomGroup` from this import, I would suggest you only import that class. I am concerned about import conflicts down the line. We used to have rules about where you can import top-level; I don't remember them, though :roll_eyes:

Reply:
Good point, the rule was that nothing in lib should import anything from MDA core... so in that sense the whole PR goes against that idea.

Reply:
This is a circular import unless done at the top level. You can always use a top level import to get around the circular-ness but whether it is good practice is another question. 

The circular import occurs between `lib.util` and `core.groups`. I think this can be fixed by moving `@check_coords ` to its own module in `lib` because `groups.py` imports `lib.util` but doesn't require `check_coords`. Would people be happy with this solution?

Or we can hit it with a hammer and use a top level import. Or we can look into a different solution. Let me know what you think.

Reply:
I find the mda import here very weird and out of place, given what else is `util`. 

I'd say, if the import is required then rather move `check_coordinates()` and anything related to a separate module. (Presumably, you'd leave the old code here with a deprecation and create the new code with more functionality elsewhere until we can remove it in 3.0.). 

Although I am making a suggestion to use duck-typing in `check_coordinates()`, it might not a bad idea to just break up `util` anyway — it's become a big kitchen sink!